tests/test_parameters.py

```diff
@@ -20,9 +20,9 @@ def parameters():
                              user_data=1))
     pars.add(lmfit.Parameter(name='b', value=0.0, vary=True, min=-250.0,
                              max=250.0, expr="2.0*a", brute_step=25.0,
-                             user_data=2.5))
+                             user_data={'test': 123}))
     exp_attr_values_A = ('a', 10.0, True, -100.0, 100.0, None, 5.0, 1)
-    exp_attr_values_B = ('b', 20.0, False, -250.0, 250.0, "2.0*a", 25.0, 2.5)
+    exp_attr_values_B = ('b', 20.0, False, -250.0, 250.0, "2.0*a", 25.0, {'test': 123})
```

Comment:
what's the purpose of this change - why would one want to make things fail when `user_data` is a dictionary? Or am I missing something here?

Reply:
With this change, the test covers a case that was not covered before, where the user data is a mutable object such as a dict or list. See the example code in the PR: modifying the user data of the copied `params2` modifies the user data of the original `params`.

Reply:
I see what you mean. In the current test "user_data" was an int/float value in both cases, replacing one with a more complicated type makes sense!